core/src/main/java/org/apache/calcite/sql/SqlBinaryOperator.java

```diff
@@ -186,6 +186,10 @@ public RelDataType deriveType(
       final SqlMonotonicity mono0 = call.getOperandMonotonicity(0);
       final SqlMonotonicity mono1 = call.getOperandMonotonicity(1);
       if (mono1 == SqlMonotonicity.CONSTANT) {
+        // e.g., mono / null --> constant (null!)
```

Comment:
hmm.. do you make the change `(null!)` by intention?  What does the `!` mean ?

Reply:
Hey, @jinxing64, thanks for review. I just follow the comment style in line 197, Maybe I can make it more clear.
`// mono / zero --> constant (infinity!) `